### PR TITLE
feat!: support source map with multiple sources

### PR DESCRIPTION
**Pull request by MoLow**

### CODE REVIEW

**Comment by MoLow** [comment]
I have seen https://github.com/istanbuljs/v8-to-istanbul/pull/91 after implementing this. 
my soultion seemd a little cleaner, I have compared results of runs and they look the same

**Comment by MoLow** [comment]
@bcoe @lukeapage wdyt?

**Comment by benjamingr** [comment]
@bcoe any chance this could get a review ?

**Comment by bcoe** [comment]
@MoLow thank you for taking a crack at this, I'll make an effort to give you a more thorough review soon.

File: tap-snapshots/test-v8-to-istanbul.js-TAP.test.js [hunk 1]
```diff
@@ -1870,358 +1870,10 @@ Object {
...
-        "start": Object {
-          "column": 12,
-          "line": 20,
-        },
-      },
-      "name": "e",
-    },
-  },
+  "b": Object {},
+  "branchMap": Object {},
```
Comment (bcoe) [inline tap-snapshots/test-v8-to-istanbul.js-TAP.test.js:1874]: this seems weird to me, I wouldn't have expected us to need to update the snapshot for the non-multi-source use-case, and it looks like this updates the snapshot with empty information.

Comment (MoLow) [reply tap-snapshots/test-v8-to-istanbul.js-TAP.test.js]: @bcoe this was indeed a bug, I fixed it

**Comment by bcoe** [comment]
@MoLow sorry for the slow review, this implementation seems close to what I was expecting we'd do 👍 

Great work, and thank you @lukepage for getting this in motion.

**Comment by bcoe** [comment]
Hey @MoLow I've been working on dual mode modules for `yargs-parser`, so actually better understand the  problem we're trying to solve here and could give better review.

## Issues with yargs-parser

I tried running `c8` against `yargs-parser` with your branch of `v8-to-istanbul`. The results for text output are very promising:

Screen Shot 2020-08-01 at 12 31 44 PM

However, when I open an HTML report there are some issues:

Screen Shot 2020-08-01 at 12 33 46 PM

My guess is we're missing resolving a path somewhere?

## Issues with c8 test suite

I also ran the test suite against `c8` and am getting a couple additional path related failures:

Screen Shot 2020-08-01 at 12 35 03 PM

---

I bet there's a chance these issues are related, and we're just missing  a resolve somewhere.

**Comment by bcoe** [comment]
@MoLow update, I got everything debugged; your work was 99% of the way there 😄

**Pull request merged**